extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSearchExtractor.java

```diff
@@ -133,15 +353,217 @@ private void collectStreamsFrom(InfoItemsSearchCollector collector, JsonArray vi
+                if (searchType.equals(MUSIC_SONGS) || searchType.equals(MUSIC_VIDEOS)) {
+                    collector.commit(new YoutubeStreamInfoItemExtractor(info, timeAgoParser) {
+                        @Override
+                        public String getUrl() throws ParsingException {
+                            String url = getUrlFromNavigationEndpoint(info.getObject("doubleTapCommand"));
+                            if (url != null && !url.isEmpty()) return url;
+                            throw new ParsingException("Could not get url");
+                        }
+
+                        @Override
+                        public String getName() throws ParsingException {
+                            String name = getTextFromObject(info.getArray("flexColumns").getObject(0)
+                                    .getObject("musicResponsiveListItemFlexColumnRenderer").getObject("text"));
+                            if (name != null && !name.isEmpty()) return name;
+                            throw new ParsingException("Could not get name");
+                        }
+
+                        @Override
+                        public long getDuration() throws ParsingException {
+                            String duration = getTextFromObject(info.getArray("flexColumns").getObject(3)
+                                    .getObject("musicResponsiveListItemFlexColumnRenderer").getObject("text"));
+                            if (duration != null && !duration.isEmpty())
+                                return YoutubeParsingHelper.parseDurationString(duration);
+                            throw new ParsingException("Could not get duration");
+                        }
+
+                        @Override
+                        public String getUploaderName() throws ParsingException {
+                            String name = getTextFromObject(info.getArray("flexColumns").getObject(1)
+                                    .getObject("musicResponsiveListItemFlexColumnRenderer").getObject("text"));
+                            if (name != null && !name.isEmpty()) return name;
+                            throw new ParsingException("Could not get uploader name");
+                        }
+
+                        @Override
+                        public String getUploaderUrl() throws ParsingException {
+                            if (searchType.equals(MUSIC_VIDEOS)) return null;
+                            String url = getUrlFromNavigationEndpoint(info.getArray("flexColumns")
+                                    .getObject(1).getObject("musicResponsiveListItemFlexColumnRenderer")
+                                    .getObject("text").getArray("runs").getObject(0).getObject("navigationEndpoint"));
+                            if (url != null && !url.isEmpty()) return url;
+                            throw new ParsingException("Could not get uploader url");
```

Comment:
Null pointer exceptions when the uploader url is not available, a null should be returned in those cases.

Reply:
Could you tell me where you've seen that happen? I've never seen that happen, and I don't think @opusforlife2 has seen that happen either.

Reply:
@mauriciocolli Can you provide a link?

Reply:
I found one. Thanks!